modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/FriendlyURLDLFileEntryLocalServiceWrapper.java

```diff
@@ -0,0 +1,209 @@
+			File file, InputStream inputStream, long size, Date expirationDate,
+			Date reviewDate, ServiceContext serviceContext)
+		throws PortalException {
+
+		DLFileEntry dlFileEntry = super.updateFileEntry(
+			userId, fileEntryId, sourceFileName, mimeType, title, description,
+			changeLog, dlVersionNumberIncrease, fileEntryTypeId,
+			ddmFormValuesMap, file, inputStream, size, expirationDate,
+			reviewDate, serviceContext);
+
+		if (_ffFriendlyURLEntryFileEntryConfiguration.enabled()) {
+			_updateFriendlyURL(dlFileEntry, title);
+		}
+
+		return dlFileEntry;
+	}
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		_ffFriendlyURLEntryFileEntryConfiguration =
+			ConfigurableUtil.createConfigurable(
+				FFFriendlyURLEntryFileEntryConfiguration.class, properties);
+	}
+
+	private void _addFriendlyURLEntry(DLFileEntry dlFileEntry)
+		throws PortalException {
+
+		String urlTitle = _friendlyURLEntryLocalService.getUniqueUrlTitle(
+			dlFileEntry.getGroupId(),
+			_classNameLocalService.getClassNameId(FileEntry.class),
+			dlFileEntry.getFileEntryId(), dlFileEntry.getTitle(),
+			LanguageUtil.getLanguageId(LocaleUtil.getSiteDefault()));
+
+		_friendlyURLEntryLocalService.addFriendlyURLEntry(
+			dlFileEntry.getGroupId(),
+			_classNameLocalService.getClassNameId(FileEntry.class),
+			dlFileEntry.getFileEntryId(), urlTitle,
+			ServiceContextThreadLocal.getServiceContext());
+	}
+
+	private Map<String, String> _getUniqueUrlTitleMap(
+		long groupId, long fileEntryId, String title,
+		Map<String, String> urlTitleMap) {
+
+		Map<String, String> newUrlTitleMap = new HashMap<>();
+
+		for (Map.Entry<String, String> entry : urlTitleMap.entrySet()) {
+			String languageId = entry.getKey();
+
+			String urlTitle = urlTitleMap.get(languageId);
+
+			if (Validator.isNotNull(urlTitle) ||
+				((urlTitle != null) && urlTitle.equals(StringPool.BLANK))) {
+
+				urlTitle = _friendlyURLEntryLocalService.getUniqueUrlTitle(
+					groupId,
+					_classNameLocalService.getClassNameId(FileEntry.class),
+					fileEntryId, title, languageId);
+
+				newUrlTitleMap.put(languageId, urlTitle);
+			}
+		}
+
+		return newUrlTitleMap;
+	}
+
+	private void _updateFriendlyURL(DLFileEntry dlFileEntry, String title)
+		throws PortalException {
+
+		try {
+			FriendlyURLEntry friendlyURLEntry =
+				_friendlyURLEntryLocalService.getMainFriendlyURLEntry(
+					_classNameLocalService.getClassNameId(FileEntry.class),
+					dlFileEntry.getFileEntryId());
+
+			_friendlyURLEntryLocalService.updateFriendlyURLEntry(
+				friendlyURLEntry.getFriendlyURLEntryId(),
+				friendlyURLEntry.getClassNameId(),
+				friendlyURLEntry.getClassPK(),
+				friendlyURLEntry.getDefaultLanguageId(),
+				_getUniqueUrlTitleMap(
+					dlFileEntry.getGroupId(), dlFileEntry.getFileEntryId(),
+					title, friendlyURLEntry.getLanguageIdToUrlTitleMap()));
+		}
+		catch (Exception exception) {
+			_addFriendlyURLEntry(dlFileEntry);
+		}
+	}
```

Comment:
Shouldn't we update the friendly URL only if `dlFileEntry.getTitle() != title`? Here we are always creating a new one for each update, even if the title didn't change.

Reply:
Well, 
we can check if the title is != than the friendlyUrl.title from the _friendlyURLEntryLocalService.getMainFriendlyURLEntry( because if in the future we want to split the title and the friendlyURl we can change one but not another.

And we should do it after the getMainFriendlyURLEntry because if it's an old File, we would update the file but not create the friendlyURLEntry

Reply:
Thinking on it. To prevent the change on the future, I will pass the title (urlTitle) in both methods creation and update